### PR TITLE
[#5659] improvement(catalog-mysql) : support load MySQL BOOLEAN data in Trino

### PR DESCRIPTION
**Pull request by sunxiaojian**
### Why are the changes needed?

Fix: https://github.com/apache/gravitino/issues/5659

### How was this patch tested?
image

### CODE REVIEW

**Comment by sunxiaojian** [comment]
@xunliu PTAL

File: catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java [hunk 1]
```diff
@@ -43,6 +45,12 @@ public class MysqlTypeConverter extends JdbcTypeConverter {
   @Override
   public Type toGravitino(JdbcTypeBean typeBean) {
     switch (typeBean.getTypeName().toLowerCase()) {
+      case BIT:
+      case BIT_UNSIGNED:
+        if (typeBean.getColumnSize() == null || Integer.parseInt(typeBean.getColumnSize()) <= 1) {
```
Comment (tengqm) [inline catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:50]: The fact that `getColumnSize()` returns a string is a surprise to me.
Anyway.

**Comment by diqiu50** [comment]
Please enable the  boolean type of the Trino integration test for mysql catalog.  The data type of  `f7` should be `BOOLEA` 
https://github.com/apache/gravitino/blob/99a999a1ea05cf8d82df4a14788ce4ab46a52946/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00006_datatype.sql#L4-L23

File: catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java (hunk 1 above)
Comment (mchades) [reply catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:50]: > The fact that `getColumnSize()` returns a string is a surprise to me. Anyway.

The same to me.

@sunxiaojian could you plz open a new issue to track this?  For changing the `columnSize` type to an integer type.

Comment (sunxiaojian) [reply catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:50]: @mchades Okay, this should be of type int

Comment (mchades) [reply catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:50]: thx a lot. Tracked by #5784

**Comment by sunxiaojian** [comment]
> Please enable the boolean type of the Trino integration test for mysql catalog. The data type of `f7` should be `BOOLEA`
> 
> https://github.com/apache/gravitino/blob/99a999a1ea05cf8d82df4a14788ce4ab46a52946/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00006_datatype.sql#L4-L23

@diqiu50 fixed

File: trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00006_datatype.sql [hunk 2]
```diff
@@ -23,12 +23,12 @@ CREATE TABLE tb01 (
...
+INSERT INTO tb01 (f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14, f15, f16)
+VALUES ('Sample text 1', 'Text1', x'65', 123.456, 7.89, 12.34, FALSE, 1, 100, 1000, 1000, 100000, DATE '2024-01-01',
         TIME '08:00:00', TIMESTAMP '2024-01-01 08:00:00', TIMESTAMP '2024-01-01 08:00:00 UTC');
 
-INSERT INTO tb01 (f1, f2, f3, f4, f5, f6, f8, f9, f10, f11, f12, f13, f14, f15, f16)
-VALUES (NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULl);
+INSERT INTO tb01 (f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13, f14, f15, f16)
+VALUES (NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULl, NULl);
 
 select * from tb01 order by f1;
```
Comment (diqiu50) [inline trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00006_datatype.sql:34]: The `tb2` need to update too

Comment (sunxiaojian) [reply trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-mysql/00006_datatype.sql:34]: > The `tb2` need to update too

fixed

File: catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java [hunk 3]
```diff
@@ -43,6 +45,12 @@ public class MysqlTypeConverter extends JdbcTypeConverter {
   @Override
   public Type toGravitino(JdbcTypeBean typeBean) {
     switch (typeBean.getTypeName().toLowerCase()) {
+      case BIT:
+      case BIT_UNSIGNED:
+        if (typeBean.getColumnSize() == null || typeBean.getColumnSize() <= 1) {
```
Comment (diqiu50) [inline catalogs/catalog-jdbc-mysql/src/main/java/org/apache/gravitino/catalog/mysql/converter/MysqlTypeConverter.java:50]: Why is the column `size <=1`?  what's happens the size=0

**Review by diqiu50 — DISMISSED** [review]
LGTM

**Review by diqiu50 — APPROVED** [review]
LGTM

**Review by mchades — APPROVED** [review]
LGTM. Thanks for your contributions!

**Pull request merged**